File: 219-ContainsDuplicateIi/219-ContainsDuplicateIi.cpp
// Last updated: 3/30/2026, 2:13:35 PM
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int,int> hashmap;
        int n = nums.size();
        for(int index = 0 ; index < n ; index++){
            if(hashmap.find(nums[index]) != hashmap.end() && abs(index - hashmap.at(nums[index]))<=k) return true;
            hashmap[nums[index]] = index;
        }
        return false;
    }
};

File: 219-ContainsDuplicateIi/219-ContainsDuplicateIi.java
// Last updated: 3/30/2026, 2:13:41 PM
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int index = 0 ; index < nums.length ; index++){
            if(map.containsKey(nums[index]) && index - map.get(nums[index])<=k){
                System.out.println(nums[index] + map.get(nums[index]));
                return true;
            }
            map.put(nums[index],index);
        }
        return false;
    }
}